app/controllers/imports_controller.rb

```diff
@@ -32,8 +32,11 @@ def create
 
       raw_data =
         case params[:import][:source]
-        when 'gpx' then Hash.from_xml(file)
+        when 'google_semantic_history' then GoogleMaps::SemanticHistoryParser.new(import, current_user.id).call
```

Comment:
Imports controller is a part of the application responsible for uploading files via the web interface. It has nothing to do with the Watcher functionality.

Reply:
got it , everything has been reverted